linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_deny/bash/shared.sh

```diff
@@ -2,33 +2,34 @@
 
 {{{ bash_instantiate_variables("var_accounts_passwords_pam_faillock_deny") }}}
 
-SYSTEM_AUTH="/etc/pam.d/system-auth"
-PASSWORD_AUTH="/etc/pam.d/password-auth"
-FAILLOCK_CONF="/etc/security/faillock.conf"
-
-if [ $(grep -c "^\s*auth.*pam_unix.so" $SYSTEM_AUTH) > 1 ] || \
-   [ $(grep -c "^\s*auth.*pam_unix.so" $PASSWORD_AUTH) > 1 ]; then
-   echo "Skipping remediation because there are more pam_unix.so entries than expected."
-   false
+if [ -f /usr/sbin/authconfig ]; then
+    authconfig --enablefaillock --update
+elif [ -f /usr/bin/authselect ]; then
+    if [ $(authselect enable-feature with-faillock) ]; then
+        authselect apply-changes
+    else
+        # If authselect can't manage the pam files, mainly due to manual modifications, the current
+        # profile will be turned back to a inter state, preserving previously enabled features.
+        ENABLED_FEATURES=$(authselect current | tail -n+3 | awk '{ print $2 }')
+        CURRENT_PROFILE=$(authselect current -r | awk '{ print $1 }')
+        authselect select $CURRENT_PROFILE --force
```

Comment:
I think the remediation should not be reverting manual changes done on the PAM files.